### PR TITLE
Fix Bug #261 destroyProcessByID not working

### LINKED ISSUE #261: destroyProcessByID not working
Hey, I am downloading an RTMP file from a site and using a Timer that reads the size of the file from the disk every second.

Downloading works fine but it continues even after I call `destroyProccessById()` and it returns true.
In the screenshot, as you can see, the timer prints the file size that is continuously increasing, which means file is being downloaded in the background. 

Any remedy for this?
Screenshot 2023-12-26 at 9 19 08 AM

### PR DESCRIPTION
**Pull request by farimarwat**
The ProcessBuilder() starts process for Python lib. Python lib further creates a process for ffmpeg if used for downloading. So the process returned by .start() method is only capable to kill child process(python) of the app not sub process(ffmpeg). As this(ffmpeg) belongs to the app tree, so we are able to kill it even android has restrctions to kill other's app process.
Tree for process:
App Process(PID)
-Python Process(PID)
--FFMPEG Process(PID)
Now the issue was that if we use "destroyProcessById()" the FFMPEG process still continue to download the app. So we have to manually kill this as well.
This patch will solve the bug.

Close #261

### CODE REVIEW

**Comment by JunkFood02** [comment]
Thanks for your PR! According to the code, it seems we're trying to terminating all the child processes when destroy a single yt-dlp process, this might not be the expected behavior if we want to allow multiple yt-dlp instance to be running at the same time

**Comment by JunkFood02** [comment]
Maybe we can try using `pstree` or something similiar to identify the child processes of the exact yt-dlp process we want to destroy

File: library/src/main/AndroidManifest.xml
Comment (JunkFood02) [inline library/src/main/AndroidManifest.xml:1]: Do we really need this permission when killing our own processes? See also: https://developer.android.com/reference/android/Manifest.permission#KILL_BACKGROUND_PROCESSES

File: library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.kt [hunk 1]
```diff
@@ -120,6 +120,7 @@ object YoutubeDL {
 
     fun destroyProcessById(id: String): Boolean {
         if (idProcessMap.containsKey(id)) {
+            killChildProcesses()
```
Comment (JunkFood02) [inline library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.kt:123]: We should only terminate the child processes of the exact process to be destory

File: library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.kt [hunk 2]
```diff
@@ -132,8 +133,18 @@ object YoutubeDL {
             }
         }
         return false
-    }
 
+    }
+    fun killChildProcesses() {
+        val process = Runtime.getRuntime().exec("ps -A -o pid,cmd")
+        process.inputStream.bufferedReader().useLines { lines ->
+            lines.filter { it.contains("ffmpeg") }
```
Comment (JunkFood02) [inline library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.kt:141]: Maybe aria2c too? Not sure about how many external executables in our libraies could have this issue

Comment (JunkFood02) [reply library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.kt:141]: See also: https://github.com/JunkFood02/Seal/issues/180

**Review by JunkFood02 — CHANGES_REQUESTED** [review]

File: library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.kt (hunk 2 above)
Comment (JunkFood02) [reply library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.kt:141]: And this function should actually be named `killAllFFmpegProcesses` according to its implementation

**Comment by farimarwat** [comment]
I have successfully updated the code to kill only selective ffmpeg process. Kindly test and feed back me

**Pull request closed without merging**